### ISSUE TITLE
props parameter forms (parameter property, default value) are handled inconsistently across rules

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#534]
### Describe the bug

The three props-related rules handle non-plain-identifier constructor `props` parameters (parameter property `private readonly props: XProps`, default value `props: XProps = {}`) inconsistently — one produces a false positive, the others silently stop checking:

1. **construct-constructor-property — false positive**: `private readonly props: XProps` (a `TSParameterProperty`) is reported with *"must have the property names 'scope, id' or 'scope, id, props'"* even though the parameter **is** named `props`. `unwrapDefaultValue` only unwraps `AssignmentPattern`, so the `TSParameterProperty` binding falls into the `binding.type !== Identifier` branch.
2. **props-name-convention — false negative**: both `props: WrongProps = {}` and `private readonly props: WrongProps` skip the naming check entirely (`params[2].type !== Identifier` → early return), so a wrongly named props interface is never reported. Note that the default-value form is explicitly supported (valid) in construct-constructor-property since #509, so users writing that recommended style silently lose this rule.
3. **no-unused-props — false negative**: with `private readonly props: XProps`, the rule is silently disabled — genuinely unused properties are not reported.

**Repro (each class in isolation):**

```ts
import { Construct } from "constructs";

interface WrongNameProps {
  readonly sample?: string;
}

// 1. construct-constructor-property: reported although names are scope/id/props
export class ParamPropCtor extends Construct {
  constructor(scope: Construct, id: string, private readonly props: WrongNameProps) {
    super(scope, id);
  }
}

// 2. props-name-convention: interface name should be 'DefaultValueNamingProps' but nothing is reported
export class DefaultValueNaming extends Construct {
  constructor(scope: Construct, id: string, props: WrongNameProps = {}) {
    super(scope, id);
    void props;
  }
}

// 3. no-unused-props: 'neverUsed' is truly unused but nothing is reported
interface ParamPropProps {
  readonly bucketName: string;
  readonly neverUsed: string;
}

export class ParamPropConstruct extends Construct {
  constructor(scope: Construct, id: string, private readonly props: ParamPropProps) {
    super(scope, id);
    console.log(this.props.bucketName);
  }
}
```

### Expected Behavior

- 1: no error (parameter is named `props` and typed with a Props interface), or — if parameter properties are intentionally disallowed — a dedicated message saying so instead of the misleading name-convention message.
- 2: `invalidPropsName` reported for both forms.
- 3: `unusedProp` reported for `neverUsed`.

### Current Behavior

```
repro/construct-constructor-property/parameter-property.ts  19:5  error  Constructor of a class extending Construct must have the property names 'scope, id' or 'scope, id, props'  awscdk/construct-constructor-property
```

(19:5 is the `private readonly props` parameter.) props-name-convention and no-unused-props report nothing for cases 2 and 3. Oxlint behaves identically for all three cases.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), macOS, Node.js 24.18.0, rule-isolated configs.
- Sources (identical in both packages):
  - `packages/*/src/rules/construct-constructor-property.ts` — `unwrapDefaultValue` handles `AssignmentPattern` only
  - `packages/*/src/rules/props-name-convention.ts` — `propsParam?.type !== Identifier` early return
  - `packages/*/src/rules/no-unused-props/index.ts` — `getPropsParam` returns null for non-`Identifier` params (the `ObjectPattern` exclusion is documented as intentional there, but `TSParameterProperty` / `AssignmentPattern` are not)
- Whatever the intended policy for parameter properties is, the three rules should agree on it.
